### PR TITLE
new options to NameSpaceRuleChecker + minor checker-code cleanups

### LINKED ISSUE #221: Extend the NameSpaceRuleChecker to be triggered by user-specified trigger-objects
Extend the NameSpaceRuleChecker with two new (optional) options (backward compatible):

- If trigger_on_dynamic_objects (optional) is true, then the namespacerule checker performs all its checks as usual. (Default: true)

- trigger_objects (optional) is a list of strings; if this list is non-empty, any request containing any of these strings will be replayed using the attacker credentials. This way, the user can direct this checker to try attacker credentials for any request containing specific trigger_object values, like "tenantID".

### PR DESCRIPTION
**Pull request by PatGod**
Close #221 

This PR implements two new options for the NameSpaceRuleChecker (and includes some minor checker-code cleanups).

If trigger_on_dynamic_objects (optional) is true, then the namespacerule checker performs all its checks as usual. (Default: true)

trigger_objects (optional) is a list of strings; if this list is non-empty, any request containing any of these strings will be replayed using the attacker credentials. This way, the user can direct this checker to try attacker credentials for any request containing specific trigger_object values, like "tenantID".

The documentation has been updated.

Tests: manual testing with a sample RESTler job.

### CODE REVIEW

File: restler/checkers/namespace_rule_checker.py [hunk 1]
```diff
@@ -87,30 +99,53 @@ def _namespace_rule(self):
...
-        if self._mode != 'exhaustive' and not self._sequence.last_request.consumes:
-            return
+        last_request = self._sequence.last_request
+        last_rendering, last_parser = last_request.render_current(self._req_collection.candidate_values_pool)
+        
+        last_request_contains_a_trigger_object = False
+        for obj in self._trigger_objects:
+            if last_rendering.find(obj) != -1:
+                last_request_contains_a_trigger_object = True
+                self._checker_log.checker_print(f"\nThe last request contains trigger_object: {obj}\nThe last request rendering is:\n{last_rendering}\n")
```
Comment (marina-p) [inline restler/checkers/namespace_rule_checker.py:111]: >nThe [](http://example.com/codeflow?start=51&length=4)

Could you please break this over multiple lines for readability?

Comment (marina-p) [inline restler/checkers/namespace_rule_checker.py:111]: >\nThe  [](http://example.com/codeflow?start=99&length=6)

same comment as above, please break over two lines

**Review by marina-p — DISMISSED** [review]
:shipit:

**Comment by PatGod** [comment]
@marina-p Thanks! I have fixed that long line.

**Review by marina-p — APPROVED** [review]
:shipit:

**Pull request merged**